### PR TITLE
Simplify isa_builder macro

### PR DESCRIPTION
**Pull request by nbp**
This changes replace the inner function creation by a simple if using the `cfg!` macro.

### CODE REVIEW

File: cranelift-codegen/src/isa/mod.rs [hunk 1]
```diff
@@ -90,29 +90,25 @@ mod stack;
...
 macro_rules! isa_builder {
-    ($name:ident, $feature:tt) => {{
-        #[cfg(feature = $feature)]
-        fn $name(triple: Triple) -> Result<Builder, LookupError> {
-            Ok($name::isa_builder(triple))
-        };
-        #[cfg(not(feature = $feature))]
-        fn $name(_triple: Triple) -> Result<Builder, LookupError> {
-            Err(LookupError::Unsupported)
+    ($name: ident, $feature: tt, $triple: ident) => {{
```
Comment (bjorn3) [inline cranelift-codegen/src/isa/mod.rs]: The name is now unuses.

Comment (bnjbvr) [reply cranelift-codegen/src/isa/mod.rs]: It's used line 95.

Comment (bjorn3) [reply cranelift-codegen/src/isa/mod.rs]: Oops.

Comment (bnjbvr) [reply cranelift-codegen/src/isa/mod.rs]: There's probably a macro way to reuse $name for $feature, using the `stringify!` macro, though.

Comment (nbp) [reply cranelift-codegen/src/isa/mod.rs]: Doing so would imply that we always keep the same for both, and modules names are more restricted than feature names.

**Comment by nbp** [comment]
Apparently we cannot refer to a conditional module name based only on the branch condition. The code has to be valid whether or not the feature is present or not.

I will rewrite it to

```rust
#[cfg(feature = $feature)]
{
   ...
}
#[cfg(feature != $feature)]
{
   ...
}
```

Which avoids the function call.

**Pull request merged**